In [1]:
import sys
sys.path.append("./gym-examples/") # to be ale to import from the gym_examples folder, which I simply put inside this project
import gym_examples
import gym
from stable_baselines3 import SAC

In [2]:
env = gym.make('gym_examples/UR10-v0', headless=False, responsive_ui=True)

In [6]:
model = SAC.load("sac_circular")

In [8]:
cnt = 0
obs = env.reset()
done = False

first_orientation = env.target.get_orientation()
while not done:
    action, _states = model.predict(obs, deterministic=True)
    obs, reward, done, _ = env.step(action)
    cnt += 1
    env.agent_ee_tip.set_orientation(first_orientation)
    #print(f'step {cnt} | target orientation {env.target.get_orientation()}')
    #print(f'step {cnt} | tip orientation {env.agent_ee_tip.get_orientation()}')
    #print(f'step {cnt} | tip orientation relative to target {env.agent_ee_tip.get_orientation(relative_to=env.target)}')

step 1 | target orientation [-1.57081223 -1.55584574 -1.57081223]
step 1 | tip orientation [-1.57079637 -1.55149806 -1.57079637]
step 1 | tip orientation relative to target [-4.33981698e-03  1.79955819e-07 -1.16956436e-07]
step 2 | target orientation [-1.57081223 -1.55584574 -1.57081223]
step 2 | tip orientation [-1.57080865 -1.55150425 -1.57080865]
step 2 | tip orientation relative to target [-4.33969917e-03  5.94517182e-08  1.68928498e-08]
step 3 | target orientation [-1.57081223 -1.55584574 -1.57081223]
step 3 | tip orientation [-1.57079637 -1.55149806 -1.57079637]
step 3 | tip orientation relative to target [-4.33987472e-03  1.78595656e-07  2.36706619e-08]
step 4 | target orientation [-1.57081223 -1.55584574 -1.57081223]
step 4 | tip orientation [-1.57080865 -1.55150735 -1.57080865]
step 4 | tip orientation relative to target [-4.33979137e-03  1.79352710e-07 -5.36276552e-08]
step 5 | target orientation [-1.57081223 -1.55584574 -1.57081223]
step 5 | tip orientation [-1.57079637 -1.5

step 1 | target orientation [-1.57079637  0.01496714 -1.57079637]
step 1 | tip orientation [-1.56743622 -0.03082427 -1.51425421]

step 100 | target orientation [-1.57079637  0.01496714 -1.57079637]
step 100 | tip orientation [-2.63485289  0.14819838 -0.0763502 ]

step 600 | target orientation [ 1.57079625 -0.01931188  1.57079637]
step 600 | tip orientation [-1.89009655  0.46229902 -2.38717937]

step 1100 | target orientation [-1.57079637  0.01929752 -1.57079637]
step 1100 | tip orientation [-2.97047067 -0.24640548  0.54014832]

In [9]:
env.shutdown()

[CoppeliaSim:loadinfo]   done.
